src/index.js

```diff
@@ -379,6 +394,17 @@ class InputElement extends React.Component {
     let newSelection = changedState.selection;
     let newValue = changedState.value;
 
+    if (this.props.noSpaceBetweenChars && this.maskOptions.maskChar) {
+      const collapsedState = collapseSpaceBetweenChars(
```

Comment:
It seems that the value passed to `beforeMaskedValueChange` will differ from the actual input value. `beforeMaskedValueChange` is intended to be a low-level escape hatch with raw data. Although it's overcomplicated and has to be redesigned, this PR isn't the right place to do that.